cpp/src/parquet/arrow/writer.cc

```diff
@@ -604,5 +634,17 @@ Status WriteTable(const ::arrow::Table& table, ::arrow::MemoryPool* pool,
   return writer->Close();
 }
 
+Status WriteRecordBatchReader(
+    ::arrow::RecordBatchReader* reader, ::arrow::MemoryPool* pool,
+    const std::shared_ptr<::arrow::io::OutputStream>& sink,
+    const std::shared_ptr<WriterProperties>& properties,
+    const std::shared_ptr<ArrowWriterProperties>& arrow_properties) {
+  std::unique_ptr<FileWriter> writer;
+  RETURN_NOT_OK(FileWriter::Open(*reader->schema(), pool, sink, properties,
+                                 arrow_properties, &writer));
+  RETURN_NOT_OK(writer->WriteRecordBatchReader(reader));
+  return writer->Close();
+}
```

Comment:
I'd prefer to avoid proliferating new convenience APIs

Reply:
I have move /WriteRecordBatch(|Reader)/ into dataset/test_utils.h